apollo-normalized-cache/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/internal/WatcherInterceptor.kt

```diff
@@ -0,0 +1,72 @@
+package com.apollographql.apollo3.cache.normalized.internal
+
+import com.apollographql.apollo3.api.ApolloRequest
+import com.apollographql.apollo3.api.ApolloResponse
+import com.apollographql.apollo3.api.CustomScalarAdapters
+import com.apollographql.apollo3.api.Operation
+import com.apollographql.apollo3.api.Query
+import com.apollographql.apollo3.cache.normalized.ApolloStore
+import com.apollographql.apollo3.cache.normalized.api.dependentKeys
+import com.apollographql.apollo3.cache.normalized.isRefetching
+import com.apollographql.apollo3.cache.normalized.watch
+import com.apollographql.apollo3.exception.ApolloException
+import com.apollographql.apollo3.interceptor.ApolloInterceptor
+import com.apollographql.apollo3.interceptor.ApolloInterceptorChain
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.emitAll
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onCompletion
+import kotlinx.coroutines.flow.onEach
+
+internal class WatcherInterceptor(val store: ApolloStore) : ApolloInterceptor {
+  override fun <D : Operation.Data> intercept(request: ApolloRequest<D>, chain: ApolloInterceptorChain): Flow<ApolloResponse<D>> {
+    if (!request.watch) {
+      return chain.proceed(request)
+    }
+
+    check(request.operation is Query) {
+      "It's impossible to watch a mutation or subscription"
+    }
+
+    val customScalarAdapters = request.executionContext[CustomScalarAdapters]!!
+    var watchedKeys: Set<String>? = null
+    var isRefetching = false
+
+    return flowOf(flowOf(emptySet()), store.changedKeys)
+        .flattenConcatPolyfill()
```

Comment:
Is this equivalent (and clearer?)?

```suggestion
    return store.changedKeys.onStart { emit(emptySet()) }
```

Reply:
👍  much much better